zio-spark-test/src/main/scala/zio/spark/test/package.scala

```diff
@@ -20,4 +24,27 @@ package object test {
   def Dataset[T: Encoder](values: T*)(implicit trace: Trace): SIO[Dataset[T]] = values.toDataset
 
   def RDD[T: ClassTag](values: T*)(implicit trace: Trace): SIO[RDD[T]] = values.toRDD
+
+  implicit class ExpectOps[T](dataset: Dataset[T]) {
+    def expectAll(matchers: LineMatcher[T]*)(implicit trace: Trace, sourceLocation: SourceLocation): Task[TestResult] =
+      dataset.collect.map { rows =>
+        TestResult(
+          TestArrow
+            .make[Any, Boolean] { _ =>
+              val boolean = rows.forall(row => matchers.exists(_.respect(row)))
+              TestTrace.boolean(boolean)(ErrorMessage.text("One of the row does not respect any matcher"))
+            }
+            .withCode("Placeholder")
+            .withLocation(sourceLocation)
+        )
+      }
+  }
+
+  val __ = Anything
+
+  object row {
+    def apply[T](predicate: T => Boolean): ConditionalMatcher[T] = ConditionalMatcher(predicate)
+    def apply[T](value: T): DataMatcher[T]                       = DataMatcher(value)
+    def apply(value: Any, values: Any*): RowMatcher              = RowMatcher(Row(value +: values: _*))
```

Comment:
```scala
def apply(valueMatcher: ValueMatcher, valueMatcher*: ValueMatcher*):RowMatcher
```
with different ways to build a valueMatcher.
``` 
implicit def conversion[T:ToValueMatcher](t:T):ValueMatcher
implicit def keyConversion[T:ToValueMatcher](t2:(String,T)):ValueMatcher
```
with possible future syntaxes being
```
row(__, "name" -> ((_:String).isEmpty || isNull))
row(__, "adr[0]" -> row[Adr](_.countryCode == "FR)
row(__,  "name" -> (v:String) => v == null || v.isEmtpy) // could reuse at lowcost zio-test macros on assertTrue
```

Reply:
What about datasets ? You only tackle dataframes with this syntax no ?

Reply:
If we take :
`row(__, "name" -> ((_:String).isEmpty || isNull), 20)`
There are both positional and key based predicates, what it possibly means ?

Reply:
it should work on dataset too.

we could have 
```scala
expectAll(
  row(_.isAdult).allowMultipleMatches, 
  row("toto", 13))
```

Reply:
Same as in Scala, it doesn't work. We could make it work, but that is too ambiguous.
```scala
def default[T]: T => Boolean = _ => true

def checkMatch( age: Int => Boolean = default,
                surname: String=> Boolean = default,
                name: String => Boolean =default):Any = ???

def __ = ???

checkMatch(__, name = _.contains("toto"))
checkMatch(__, name = _.contains("toto"), default)
```

![image](https://user-images.githubusercontent.com/83247/210793059-e50bfb6a-ec19-49e9-b449-f4505852c7bc.png)